zerver/lib/markdown/__init__.py

```diff
@@ -89,6 +89,17 @@ class LinkInfo(TypedDict):
     remove: Optional[Element]
 
 
+class MessageRenderingResult(TypedDict):
```

Comment:
Why a TypedDict rather than an @dataclass?  With `@dataclass`, we can use the `rendering_result.rendered_content` syntax, which is both simpler and also hints that this is a typed object.